web_src/js/webcomponents/dir-auto.js

```diff
@@ -0,0 +1,12 @@
+const observer = new MutationObserver((mutationList) => {
+  for (const mutation of mutationList) {
+    for (const el of mutation.addedNodes) {
+      if (el.nodeType === Node.ELEMENT_NODE || el.nodeType === Node.DOCUMENT_FRAGMENT_NODE) {
+        for (const child of el.querySelectorAll('input:not([dir]), textarea:not([dir])')) {
+          child.setAttribute('dir', 'auto');
+        }
+      }
+    }
+  }
+});
+observer.observe(document, {subtree: true, childList: true});
```

Comment:
```suggestion
const excludeTypes = new Set(['hidden', 'checkbox']);

(new MutationObserver((mutationList) => {
  for (const mutation of mutationList) {
    for (const el of mutation.addedNodes) {
      if (el.nodeType === Node.ELEMENT_NODE || el.nodeType === Node.DOCUMENT_FRAGMENT_NODE) {
        for (const child of el.querySelectorAll('input:not([dir]), textarea:not([dir])')) {
          if (excludeTypes.has(child.type)) continue;
          child.setAttribute('dir', 'auto');
        }
      }
    }
  }
})).observe(document, {subtree: true, childList: true});
```

Reply:
I think it needs more, see 93abd9c094d3f3e634359ec7c1d0a7f39d37c1f5